src/internal/_functionName.js

```diff
@@ -0,0 +1,5 @@
+module.exports = function _functionName(f) {
+  // String(x => x) evaluates to "x => x", so the pattern may not match.
+  var match = String(f).match(/^function (\w*)/);
+  return match == null ? '' : match[1];
```

Comment:
could you use `constructor.name` _if present_ and otherwise evaluate the stringified function?